reference/solver/batch_cg_kernels.cpp

```diff
@@ -0,0 +1,108 @@
+// SPDX-FileCopyrightText: 2017 - 2024 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include "core/solver/batch_cg_kernels.hpp"
+
+
+#include "core/solver/batch_dispatch.hpp"
+
+
+namespace gko {
+namespace kernels {
+namespace reference {
+
+
+/**
+ * @brief The batch Cg solver namespace.
+ *
+ * @ingroup batch_cg
+ */
+namespace batch_cg {
+
+
+namespace {
+
+
+constexpr int max_num_rhs = 1;
+
+
+#include "reference/base/batch_multi_vector_kernels.hpp.inc"
+#include "reference/matrix/batch_csr_kernels.hpp.inc"
+#include "reference/matrix/batch_dense_kernels.hpp.inc"
+#include "reference/matrix/batch_ell_kernels.hpp.inc"
+#include "reference/solver/batch_cg_kernels.hpp.inc"
+
+
+}  // unnamed namespace
+
+
+template <typename T>
+using settings = gko::kernels::batch_cg::settings<T>;
+
+
+template <typename ValueType>
+class kernel_caller {
+public:
+    kernel_caller(std::shared_ptr<const DefaultExecutor> exec,
+                  const settings<remove_complex<ValueType>> settings)
+        : exec_{std::move(exec)}, settings_{settings}
+    {}
+
+    template <typename BatchMatrixType, typename PrecType, typename StopType,
+              typename LogType>
+    void call_kernel(
+        const LogType& logger, const BatchMatrixType& mat, PrecType prec,
+        const gko::batch::multi_vector::uniform_batch<const ValueType>& b,
+        const gko::batch::multi_vector::uniform_batch<ValueType>& x) const
+    {
+        using real_type = typename gko::remove_complex<ValueType>;
+        const size_type num_batch_items = mat.num_batch_items;
+        const auto num_rows = mat.num_rows;
+        const auto num_rhs = b.num_rhs;
+        if (num_rhs > max_num_rhs) {
+            GKO_NOT_IMPLEMENTED;
+        }
+
+        const size_type local_size_bytes =
+            gko::kernels::batch_cg::local_memory_requirement<ValueType>(
+                num_rows, num_rhs) +
+            PrecType::dynamic_work_size(num_rows,
+                                        mat.get_single_item_num_nnz()) *
+                sizeof(ValueType);
```

Comment:
why does this return the size in elements, and the other one the size in bytes? IMO both should return the same.